components/org.wso2.carbon.identity.application.authenticator.basicauth/src/test/java/org/wso2/carbon/identity/application/authenticator/basicauth/BasicAuthenticatorTestCase.java

```diff
@@ -356,23 +439,39 @@ public void processAuthenticationResponseTestcaseWithuserStoreException() throws
         };
     }
 
-    @Test(dataProvider = "statusProvider")
-    public void initiateAuthenticationRequestTestcaseWithUnknownErrorCode(boolean statusProvider) throws
+    @DataProvider(name = "enableStatusProvider")
+    public Object[][] getEnabledOption() {
+
+        return new String[][]{
+                {"true", "false"},
+                {"false", null},
```

Comment:
what about other combinations?
like null, null

Reply:
First argument is used for checking isEnable condition. Even though it is String it is used as boolean so in that case there is no possibility for null.